gammapy/stats/tests/test_feldman_cousins.py

```diff
@@ -0,0 +1,153 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+import numpy as np
+from numpy.testing import assert_allclose, assert_raises
+from astropy.tests.helper import pytest
+
+try:
+    import scipy
+    HAS_SCIPY = True
+except ImportError:
+    HAS_SCIPY = False
+
+from ...stats import (
+    fc_find_acceptance_region_gauss,
+    fc_find_acceptance_region_poisson,
+    fc_construct_acceptance_intervals_pdfs,
+    fc_get_limits,
+    fc_fix_limits,
+    fc_find_limit,
+    fc_find_average_upper_limit,
+    fc_construct_acceptance_intervals,
+)
+
+
+@pytest.mark.skipif('not HAS_SCIPY')
+def test_acceptance_region_gauss():
+
+    sigma   = 1
+    n_sigma = 10
+    n_step  = 1000
+    cl      = 0.90
+
+    x_bins  = np.linspace(-n_sigma*sigma, n_sigma*sigma, n_step, endpoint=True)
+
+    # The test reverses a result from the Feldman and Cousins paper. According
+    # to Table X, for a measured value of 2.6 the 90% confidence interval should
+    # be 1.02 and 4.24. Reversed that means that for mu=1.02, the acceptance
+    # interval should end at 2.6 and for mu=4.24 should start at 2.6.
+    (x_min, x_max) = fc_find_acceptance_region_gauss(1.02, sigma, x_bins, cl)
+    assert_allclose(x_max, 2.6, 1e-07, 0.1)
+
+    (x_min, x_max) = fc_find_acceptance_region_gauss(4.24, sigma, x_bins, cl)
+    assert_allclose(x_min, 2.6, 1e-07, 0.1)
+
+    # At mu=0, confidence interval should start at the negative x_bins range.
+    (x_min, x_max) = fc_find_acceptance_region_gauss(0, sigma, x_bins, cl)
+    assert_allclose(x_min, -n_sigma*sigma)
+
+    # Pass too few x_bins to reach confidence level.
+    x_bins = np.linspace(-sigma, sigma, n_step, endpoint=True)
+    assert_raises(ValueError, fc_find_acceptance_region_gauss, 0, 1, x_bins, cl)
```

Comment:
This currently fails in one of the travis-ci builds:
https://travis-ci.org/gammapy/gammapy/jobs/86494765#L1507

Instead of using `numpy.testing.assert_raises`, please use `pytest.raises` (see [here](http://pytest.org/latest/builtin.html#pytest.raises)):
```python
with pytest.raises(ValueError):
    fc_find_acceptance_region_gauss(0, 1, x_bins, cl)
```
You can also assert things about the exception (see [example](https://github.com/astropy/astropy/blob/c7099d60a1d38b6074801d528ad069f4d06a8202/astropy/nddata/tests/test_flag_collection.py#L28)), but I don't think that's needed here.